### PR TITLE
fix(deps): update vulnerable `test-exclude` to v7

### PR DESCRIPTION
**Pull request by AriPerkkio**
### Description

- Fixes https://github.com/vitest-dev/vitest/discussions/5096
- Ref. https://github.com/istanbuljs/test-exclude/releases/tag/test-exclude-v7.0.0
- Ref. https://github.com/istanbuljs/test-exclude/issues/53

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Comment by AriPerkkio** [comment]
I wasn't expecting test failures 🤔 

Looks like there are some changes in the end locations of coverage maps. It shouldn't be that big of an issue but I'll need to check those manually first.

**Comment by AriPerkkio** [comment]
This breaks coverage completely on Windows. Coverage report is completely empty: https://github.com/vitest-dev/vitest/actions/runs/9445376953/job/26012848785?pr=5867

**Comment by AriPerkkio** [comment]
> This breaks coverage completely on Windows. Coverage report is completely empty: https://github.com/vitest-dev/vitest/actions/runs/9445376953/job/26012848785?pr=5867

Fixed by `test-exclude@7.0.1` ✅

File: test/coverage-test/coverage-report-tests/__snapshots__/istanbul.report.test.ts.snap [hunk 1]
```diff
@@ -2518,7 +2518,7 @@ exports[`istanbul json report 1`] = `
       "0": {
         "loc": {
           "end": {
-            "column": null,
```
Comment (sheremet-va) [inline test/coverage-test/coverage-report-tests/__snapshots__/istanbul.report.test.ts.snap:2521]: I don't understand. How can a change to file filtering affect coverage columns? 🤔

Comment (AriPerkkio) [reply test/coverage-test/coverage-report-tests/__snapshots__/istanbul.report.test.ts.snap:2521]: Not sure. This is the before-and-after:

- No visibile changes to end users
- No changes in coverage percentages

image

Comment (AriPerkkio) [reply test/coverage-test/coverage-report-tests/__snapshots__/istanbul.report.test.ts.snap]: Looks like `testExclude.glob()` returns the files in different order now. Adding a simple `.sort()` restores the old behaviour.

Why branch endings are mapped differently by `istanbul-lib-coverage` or `istanbul-lib-source-maps` based on the file order - no idea. 🤷‍♂️

**Pull request merged**